Assets/HoloToolkit/Utilities/Scripts/AdaptiveQuality/AdaptiveViewport.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+using System;
+using UnityEngine;
+
+namespace HoloToolkit.Unity
+{
+	/// <summary>
+	/// Changes the VR viewport to correlate to the requested quality, trying to maintain a steady framerate by reducing the amount of pixels rendered.
+	/// Uses the AdaptiveQuality component to respond to quality change events.
+	/// At MaxQualityLevel, the viewport will be set to 1.0 and will linearly drop of to MinViewportSize at MinQualityLevel
+	/// </summary>
+
+	public class AdaptiveViewport : MonoBehaviour
+	{
+		[SerializeField]
+		private int MaxQualityLevel = 5;
```

Comment:
The thinking here was that the AdaptiveQuality can report numbers in any range and the AdaptiveViewport can be set to only change on a subset of them. Other systems can pick adapt to values that the AdaptiveViewport is not handling. Say at quality -5, you reduce particle count, but don't change the viewport size.
Ill make a better explanation of this in the file!